ToolBox/Benzinga/BenzingaDataConverter.cs

```diff
@@ -0,0 +1,346 @@
+                instance.Symbols = new List<BenzingaSymbolData>();
+                instance.Metadata = new BenzingaMetadata()
+                {
+                    IsPro = item["bz:type"]["@pro"].ToString() == "1",
+                    FirstRun = item["bz:type"]["@pro"].ToString() == "1",
+                    Kind = item["bz:type"]["#text"].ToString()
+                };
+
+                foreach (var ticker in tickers)
+                {
+                    var sentiment = ticker["@sentiment"] == null ? (decimal?)null : Parse.Decimal(ticker["@sentiment"].ToString());
+                    var exchange = ticker["@exchange"] == null ? string.Empty : ticker["@exchange"].ToString();
+
+                    if (ticker["#text"] == null)
+                    {
+                        continue;
+                    }
+
+                    // Tickers with dots in them like BRK.A and BRK.B appear as BRK-A and BRK-B in Benzinga data.
+                    var symbolTicker = ticker["#text"].ToString().Trim().Replace('-', '.');
+                    var mappedSymbol = _mapFileResolver.ResolveMapFile(symbolTicker, instance.PublicationDate).GetMappedSymbol(instance.PublicationDate);
+
+                    if (string.IsNullOrWhiteSpace(mappedSymbol))
+                    {
+                        Log.Error($"BenzingaDataConverter.Process(): Failed to map old ticker {symbolTicker}. New ticker is null");
+                        continue;
+                    }
+
+                    instance.Symbols.Add(new BenzingaSymbolData()
+                    {
+                        Exchange = exchange,
+                        Sentiment = sentiment,
+                        Symbol = new Symbol(
+                            SecurityIdentifier.GenerateEquity(symbolTicker, Market.USA, mapSymbol: true, mapFileProvider: _mapFileProvider, mappingResolveDate: instance.PublicationDate),
+                            mappedSymbol
+                        )
+                    });
+                }
+
+                yield return instance;
+            }
+        }
+
+        /// <summary>
+        /// Writes the processed data into their respective files
+        /// </summary>
+        /// <param name="news">Enumerable containing news articles</param>
+        /// <param name="date">Date to convert for</param>
+        private void WriteToFile(IEnumerable<BenzingaNews> news, DateTime date)
+        {
+            var finalDirectory = new DirectoryInfo(Path.Combine(_destinationDirectory.FullName, "content", $"{date:yyyMMdd}"));
+            finalDirectory.Create();
+
+            foreach (var article in news)
+            {
+                var createdReference = false;
+
+                // For each ticker, add a reference to the article we're going to store under the "content" folder
+                foreach (var ticker in article.Symbols)
+                {
+                    // Invalid character in path
+                    if (ticker.Symbol.Value.Contains(":"))
```

Comment:
I think this invalid symbol is going to be stored with the article? Maybe we should filter it before so that it doesn't end up being shipped and consumed
Probably should also apply for the exchange check bellow?

Reply:
Is this common? Do we want to provide a mapping -- map to disk and map back when reading in? Just wondering what and how much data we're losing by performing this filtering step -- _also_ -- perhaps we should do an `OS.IsWindows` check here since *nix systems don't give two shakes about colons in paths.